nativelib/src/main/resources/posix/netdb.c

```diff
@@ -0,0 +1,100 @@
+#include "netdb.h"
+#include "sys/socket_conversions.h"
+#include <stddef.h>
+#include <stdlib.h>
+#include <string.h>
+
+int scalanative_getnameinfo(struct scalanative_sockaddr *addr,
+                            socklen_t addrlen, char *host, socklen_t hostlen,
+                            char *serv, socklen_t servlen, int flags) {
+    struct sockaddr *converted_addr;
+    scalanative_convert_sockaddr(addr, &converted_addr, &addrlen);
+    int status = getnameinfo(converted_addr, addrlen, host, hostlen, serv,
+                             servlen, flags);
+    free(converted_addr);
+    return status;
+}
+
+void scalanative_convert_scalanative_addrinfo(struct scalanative_addrinfo *in,
+                                              struct addrinfo *out) {
+    out->ai_flags = in->ai_flags;
+    out->ai_family = in->ai_family;
+    out->ai_socktype = in->ai_socktype;
+    out->ai_protocol = in->ai_protocol;
+    out->ai_addrlen = in->ai_addrlen;
+    out->ai_addr = NULL; // TODO
+    if (in->ai_canonname == NULL) {
+        out->ai_canonname = NULL;
+    } else {
+        out->ai_canonname = strdup(in->ai_canonname);
+    }
+    out->ai_next = NULL;
```

Comment:
This function is only used for converting the `hints` parameter and `getaddrinfo` doesn't care about `ai_next` there. Should we convert it anyway?